src/Interpreters/DDLTask.cpp

```diff
@@ -200,6 +201,14 @@ void DDLTaskBase::parseQueryFromEntry(ContextPtr context)
     ParserQuery parser_query(end, settings.allow_settings_after_format_in_insert);
     String description;
     query = parseQuery(parser_query, begin, end, description, 0, settings.max_parser_depth, settings.max_parser_backtracks);
+    if (auto * query_drop = query->as<ASTDropQuery>())
```

Comment:
It is a bit ugly to put handling for a super-special case into a generic handler function (`DDLTaskBase::parseQueryFromEntry()`).

Reply:
I will resubmit a PR to address these issues.